src/components/Dashboard.jsx

```diff
@@ -1,7 +1,206 @@
-import React from "react";
+import React, { useContext, useEffect, useState } from "react";
+import { UserContext } from "../context/UserContext";
+import { useRef } from "react";
 
 const Dashboard = () => {
-  return <div className="container">Dashboard</div>;
+  const [postList, setPostList] = useState([]);
+  const textareaRef = useRef();
+  const [postTitle, setPostTitle] = useState("");
+  const { name } = useContext(UserContext);
+
+  useEffect(() => {
+    let retrievedPosts;
+    if (localStorage.getItem("posts")) {
+      retrievedPosts = JSON.parse(localStorage.getItem("posts"));
+      setPostList(retrievedPosts);
+    }
+  }, []);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const newPost = {
+      id: new Date().getTime(),
+      title: postTitle,
+      date: new Date(),
+      author: name,
+      isPostOpen: false,
+      likeCount: 0,
+      isLiked: false,
+    };
+
+    if (postTitle) {
+      postList.push(newPost);
+      localStorage.setItem("posts", JSON.stringify(postList));
+    }
+
+    setPostTitle("");
+    textareaRef.current.focus();
+  };
+
+  const handleDelete = (id) => {
+    const deletedItem = postList.filter((item) => item.id !== id);
+    setPostList(deletedItem);
+    localStorage.setItem("posts", JSON.stringify(deletedItem));
+  };
+
+  const openCommentBox = (id) => {
+    const newPosts = [...postList];
+    const curPost = postList.find((post) => post.id === id);
+    curPost.isPostOpen = !curPost.isPostOpen;
```

Comment:
we are toggling here but i don't see it being used anywhere

Reply:
It's used in 137 number line.
`<button
                    onClick={() => openCommentBox(id)}
                    type="button"
                    className="button-comment"
                  >
                    Comment
                  </button>`

Reply:
@mkh-wtag bhai, this `curPost` is updating but we are not using or returning this value